stdlib/http/src/main/java/org/ballerinalang/net/http/nativeimpl/connection/Respond.java

```diff
@@ -108,14 +113,30 @@ public static Object nativeRespond(ObjectValue connectionObj, ObjectValue outbou
             }
         } catch (ErrorValue e) {
             unBlockStrand(strand);
-            log.debug(e.getPrintableStackTrace(), e);
-            return e;
+            if (log.isDebugEnabled()) {
+                log.debug(e.getPrintableStackTrace(), e);
+            }
+            if (connectionObj.getNativeData(WebSocketConstants.WEBSOCKET_HANDSHAKER) != null) {
+                throw getWebSocketException(outboundResponseObj);
+            } else {
+                return e;
+            }
         } catch (Throwable e) {
             unBlockStrand(strand);
-            //Exception is already notified by http transport.
-            String errorMessage = "Couldn't complete outbound response";
-            log.debug(errorMessage, e);
-            return HttpUtil.createHttpError(errorMessage, HttpErrorType.GENERIC_LISTENER_ERROR);
+            String errorMessage = "";
+            if (connectionObj.getNativeData(WebSocketConstants.WEBSOCKET_HANDSHAKER) != null) {
+                if (log.isDebugEnabled()) {
+                    log.debug(message, e);
+                }
+                throw getWebSocketException(outboundResponseObj);
```

Comment:
Throwing WebSocket exception during the HTTP flow seems incorrect. Still, the upgrade has not happed to WebSocket and this is a response flow to the HTTP request. 401 should go to the client without an exception thrown IMO

Reply:
According to the current implementation, Success and Errors are handled by [OnUpgradeResourceCallback](https://github.com/ballerina-platform/ballerina-lang/blob/master/stdlib/http/src/main/java/org/ballerinalang/net/http/websocket/server/OnUpgradeResourceCallback.java) during the Websocket upgrade connection. So If we will throw the exception, `OnError` can handle that.

Reply:
Had a call with @kalaiyarasiganeshalingam and cleared the flow. (Ignore the above comment)

It seems like from the Auth filter an HTTP response(401) is sent to the WSClient instead of WebSocket response. 
IMO at AuthFilter level, this should be handled not in the Respond method. If the Auth filter can identify the WebSocket upgrade request and respond appropriately then this exception will not have occurred. 

@ldclakmal @daneshk WDYT?

Reply:
In HTTP filter level, I don't think we have a way to identify whether this is an WS upgrade request. IMO, the transport pipeline should get updated to WS after authentication filters. Then we can simply respond (HTTP responses) with 401 or 403. Seems like, here the transport get updated before engaging filters. Shall we schedule a meeting and discuss this? WDYT?

Reply:
+1 Let's have a meeting and discuss